apps/crn-server/test/entities/event.test.ts

```diff
@@ -34,21 +34,21 @@ describe('events entity', () => {
         `Event (example) doesn't have a calendar`,
       );
     });
-    test(`throws when no team in speaker list is provided`, () => {
+    test('remove the speaker from speakers list when it does not have the team assigned', () => {
+      const squidexSpeaker = getSquidexGraphqlEventSpeakerWithUser();
+      squidexSpeaker.team = [];
+
       const event = {
         ...graphqlEvent,
         flatData: {
           ...graphqlEvent.flatData,
-          speakers: [
-            {} as NonNullable<
-              EventContentFragment['flatData']['speakers']
-            >[number],
-          ],
+          speakers: [squidexSpeaker],
         },
       };
-      expect(() => parseGraphQLEvent(event)).toThrowError(
-        'Either team or external author is required',
-      );
+      expect(parseGraphQLEvent(event)).toEqual({
+        ...getEventResponse(),
+        speakers: [],
```

Comment:
maybe add another speaker with a team and then expect only that user in the response? that way you are testing filtering them out, whereas this technically still leaves a lot of room for error (ie - if any user is missing a team it could always return an empty array)